cluster/ampagent/cmd/install.go

```diff
@@ -326,43 +327,120 @@ func createInitialNetworks() error {
 		// Check if network already exists
-		exists, err := Docker.NetworkExists(network)
+		id, err := Docker.NetworkID(network)
 		if err != nil {
 			return err
 		}
-		if !exists {
-			log.Println("Skipping non existing network:", network)
-			return nil
+		if id == "" {
+			// Skipping non existent network
+			continue
 		}
 
 		// Remove network
-		time.Sleep(2 * time.Second)
-		if err := Docker.RemoveNetwork(network); err != nil {
+		if err := Docker.RemoveNetwork(id); err != nil {
 			return err
 		}
-		log.Println("Successfully removed network:", network)
+		log.Printf("Successfully removed network %s [%s]", network, id)
+	}
+	return nil
+}
+
+func removeExitedContainers(timeout int) error {
+	i := 0
+	dontKill := []string{"amp-agent", "amp-local"}
+	var containers []types.Container
+	if timeout == 0 {
+		// default value
+		timeout = 30
+	}
+	log.Println("waiting for all services to clear up...")
+	filter := filters.NewArgs()
+	filter.Add("is-task", "true")
+	filter.Add("label", "io.amp.role=infrastructure")
+	for i < timeout {
+		containers, err := Docker.GetClient().ContainerList(context.Background(), types.ContainerListOptions{All: true, Filters: filter})
+		if err != nil {
+			return err
+		}
+		if len(containers) == 0 {
+			log.Println("cleared up")
+			break
+		}
+		for _, c := range containers {
+			switch c.State {
+			case "exited":
+				log.Printf("Removing container %s [%s]\n", c.Names[0], c.Status)
+				err := Docker.GetClient().ContainerRemove(context.Background(), c.ID, types.ContainerRemoveOptions{})
+				if err != nil {
+					if strings.Contains(err.Error(), "already in progress") {
+						// leave it to Docker
+						continue
+					} else {
+						return err
+					}
+				}
+			case "removing":
+			case "running":
+				// ignore it, it will be killed after the loop
+			default:
+				// this is not expected
+				log.Printf("Container %s found in status %s, %s\n", c.Names[0], c.Status, c.State)
+			}
+		}
+		i += 1
+		time.Sleep(1 * time.Second)
+	}
+	containers, err := Docker.GetClient().ContainerList(context.Background(), types.ContainerListOptions{All: true, Filters: filter})
+	if err != nil {
+		return err
+	}
+	if i == timeout {
+		log.Println("timing out")
+		log.Printf("%d containers left\n", len(containers))
+	}
+	//
+	for _, c := range containers {
+		for _, e := range dontKill {
+			if strings.Contains(c.Names[0], e) {
+				continue
+			}
+		}
+		log.Printf("Force removing container %s [%s]", c.Names[0], c.State)
+		err := Docker.GetClient().ContainerRemove(context.Background(), c.ID, types.ContainerRemoveOptions{Force: true})
+		if err != nil {
+			if strings.Contains(err.Error(), "already in progress") {
+				// leave it to Docker
+				continue
+			} else {
+				return err
+			}
+		}
 	}
 	return nil
 }
 
 const ampVolumesPrefix = "amp_"
 
-func removeVolumes() error {
+func removeVolumes(timeout int) error {
+	// volume remove timeout (sec)
+	if timeout == 0 {
+		// default value
```

Comment:
Prefer comment to the right of the instruction for readability:
`timeout = 5 // default timeout`